server/campaign_step_resources.py

```diff
@@ -0,0 +1,86 @@
+from typing import List, Any
+from functools import wraps
+from models.campaigns import CampaignModel, campaign_schema, campaigns_schema
+from models.campaign_steps import StepModel, step_schema, steps_schema
+from  models.prospect_step_join import ProspectStepModel, prospect_step_schema, prospect_steps_schema
+from models.prospects import ProspectModel, prospect_schema
+from models.users import UserModel, user_schema, users_schema
+from resources import login_required
+from flask_restful import Resource
+from flask import request, session
+import io, csv, redis, json
+
+
+class StepCreate(Resource):
+    def post(self):
+        data = step_schema.load(request.json)
+        # Check if step already exists
+        if StepModel.find_by_campaign(data['step']):
+            return {'message': 'step already exists'}, 409
+
+        new_step = StepModel(
+            step_name = data['step_name'],
+            email_subject = data['email_subject'],
+            email_body = data['email_body'],
+            campaign_id = data['campaign_id'],
+        )
+
+        try:
+            new_step.save_to_db()
+            responseObject = {
+                'status': 'success',
+                'message': 'Step was created',
+                'step': {
+                    'id': new_step.id,
+                    'email_subject': new_step.email_subject,
+                    'email_body': new_step.email_body,
+                    'campaign_id': new_step.campaign_id,
+                }
+            }
+            
+            return responseObject, 201
+        except Exception as e:
+            responseObject = {
+                'status': 'fail',
+                'message': 'Something went wrong. Please try again.'
+            }
+            return responseObject, 500
+
+class StepUpdate(Resource):
+    def post(self):
+        data = request.json
+        step = StepModel.find_by_campaign(data['campaign_id'])
+        StepModel.update_step(step)
```

Comment:
Doesn't this just update the step to be the same as it was before? i could be thinking about this wrong though

Reply:
You're 100% right; fixed now.